app/models/application_choice.rb

```diff
@@ -1,4 +1,6 @@
 class ApplicationChoice < ApplicationRecord
+  before_create :set_vendor_id
```

Comment:
If this is the primary key, can we replace the `id` key with this? Otherwise it wouldn't have a function.

I do wonder how Rails will treats this if we create models that reference ApplicationChoices, will it use `vendor_id` automatically?

If we don't replace `id`, perhaps we should think about the `vendor_id` name - I think it implies that it references a the ID of a Vendor, not an ApplicationChoice. No good alternatives though! Sorry.

Reply:
I was planning on trying to replace id. If we are happy with that. 👍
 
Not sure about references, I think it would use the `vendor_id`.

Yes the `vendor_id` name is pretty poor. Was just grabbing a name to get going.

Reply:
Ok cool, let's try to replace `id`! Then we don't have to bother with the name anyway.